### PR TITLE
Add Code Examples & some other tiny fixes

### PR DESCRIPTION
**Pull request by andreeaflorescu**
### Summary of the PR

This PR adds code examples for public structures (where that makes sense). It also adds some tiny fixes, more details in individual commit messages.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 [package]
 name = "vm-allocator"
 version = "0.1.0"
+description = "Helpers for allocating resources needed during the lifetime of a VM."
```
Comment (alexandruag) [inline Cargo.toml:4]: Hmm, just curious, are these actually required before publishing? (i.e. it won't work without them)

File: src/allocation_engine/DESIGN.md [hunk 2]
```diff
@@ -15,38 +15,38 @@ starting from the end of the managed address space.
...
+try to insert the range at the exact position described by the constraint,
+otherwise it will return an error.
 
 Struct `Constraint` is used to describe the overall information of the resource
-needed to be allocated. This structure is also used by IntervalTree to know where
-and how to allocate the resource. The fields that are mandatory for allocating
-a new memory slot are size of the slot, alignment of the slot and allocation policy.
-Optionally the user can specify a range where the allocator will place the allocated
-memory slot.
+needed to be allocated. This structure is also used by IntervalTree to know
```
Comment (alexandruag) [inline src/allocation_engine/DESIGN.md:23]: `IntervalTree`?

File: Cargo.toml (hunk 1 above)
Comment (andreeaflorescu) [reply Cargo.toml:4]: I am not sure about this particular one, but all rust-vmm crates have it.

Comment (lauralt) [reply Cargo.toml:4]: Yes the `description` is required: https://doc.rust-lang.org/cargo/reference/manifest.html#the-description-field.

**Review by gsserge — APPROVED** [review]

**Review by alexandruag — APPROVED** [review]

**Pull request merged**